api_app/db/repositories/base.py

```diff
@@ -10,52 +11,56 @@
 
 
 class BaseRepository:
-    def __init__(self, client: CosmosClient, container_name: str = None) -> None:
-        self._client: CosmosClient = client
-        self._container: ContainerProxy = self._get_container(container_name)
+    @classmethod
+    async def create(cls, client: CosmosClient, container_name: str = None):
+        cls = BaseRepository()
```

Comment:
`BaseRepository` in my opinion should have been an abstract class originally, but this change is not related directly to support async calls to cosmos db so maybe we refactor that in a separate issue.
Consider removing this initialization as this Base class shouldn't be instantiated directly.

Reply:
Fixed, and as discussed offline, will open a separate issue